x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/components/tables/meta_engines_table.test.tsx

```diff
@@ -0,0 +1,224 @@
+    it('passes the loading prop', () => {
+      resetMocks();
+      const wrapper = mountWithIntl(<MetaEnginesTable {...props} loading />);
+
+      expect(wrapper.find(EuiBasicTable).prop('loading')).toEqual(true);
+    });
+  });
+
+  describe('noItemsMessage', () => {
+    it('passes the noItemsMessage prop', () => {
+      resetMocks();
+      const wrapper = mountWithIntl(<MetaEnginesTable {...props} noItemsMessage={'No items.'} />);
+      expect(wrapper.find(EuiBasicTable).prop('noItemsMessage')).toEqual('No items.');
+    });
+  });
+
+  describe('actions', () => {
+    it('will hide the action buttons if the user cannot manage/delete engines', () => {
+      resetMocks();
+      const wrapper = shallow(<MetaEnginesTable {...props} />);
+      const tableRow = wrapper.find(EuiTableRow).first();
+
+      expect(tableRow.find(EuiIcon)).toHaveLength(0);
+    });
+
+    describe('when the user can manage/delete engines', () => {
+      let wrapper: ReactWrapper<any>;
+      let tableRow: ReactWrapper<any>;
+      let actions: ReactWrapper<any>;
+
+      beforeEach(() => {
+        resetMocks();
+        setMockValues({
+          ...DEFAULT_VALUES,
+          myRole: {
+            canManageMetaEngines: true,
+          },
+        });
+
+        wrapper = mountWithIntl(<MetaEnginesTable {...props} />);
+        tableRow = wrapper.find(EuiTableRow).first();
+        actions = tableRow.find(EuiIcon);
+        EnginesLogic.mount();
+      });
+
+      it('renders a manage action', () => {
+        jest.spyOn(TelemetryLogic.actions, 'sendAppSearchTelemetry');
+        jest.spyOn(KibanaLogic.values, 'navigateToUrl');
+        actions.at(1).simulate('click');
+
+        expect(TelemetryLogic.actions.sendAppSearchTelemetry).toHaveBeenCalled();
+        expect(KibanaLogic.values.navigateToUrl).toHaveBeenCalledWith('/engines/test-engine');
+      });
+
+      describe('delete action', () => {
+        it('shows the user a confirm message when the action is clicked', () => {
+          jest.spyOn(global, 'confirm' as any).mockReturnValueOnce(true);
+          actions.at(2).simulate('click');
+          expect(global.confirm).toHaveBeenCalled();
+        });
+
+        it('clicking the action and confirming deletes the engine', () => {
+          jest.spyOn(global, 'confirm' as any).mockReturnValueOnce(true);
+          jest.spyOn(EnginesLogic.actions, 'deleteEngine');
+
+          actions.at(2).simulate('click');
+
+          expect(onDeleteEngine).toHaveBeenCalled();
+        });
+
+        it('clicking the action and not confirming does not delete the engine', () => {
+          jest.spyOn(global, 'confirm' as any).mockReturnValueOnce(false);
+          jest.spyOn(EnginesLogic.actions, 'deleteEngine');
+
+          actions.at(2).simulate('click');
+
+          expect(onDeleteEngine).toHaveBeenCalledTimes(0);
+        });
+      });
+    });
+  });
+
+  describe('source engines', () => {
+    describe('source engine table', () => {
```

Comment:
Nice, I really like these tests! I do think we could lose 1 of the `describe`s as there doesn't seem to be any other tests that would require an extra level of nesting (and source engine tests are covered by another file).

```suggestion
  describe('expanded source engines', () => {
```